File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        int r=0;
        int l=0;
        int count =0;
        int curProd=1;
        for(r=0 ; r < nums.length ; r++){
            curProd*=nums[r];
            while(l<r && curProd >= k){
                curProd/=nums[l];
                l++;
            }
            if(curProd < k)
             count+=(r-l+1);

        }
        return count;
        
    }
}

File: 0713-subarray-product-less-than-k/0713-subarray-product-less-than-k.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        c=0
        l=0
        r=0
        prod=1
        for r in range(len(nums)):
            prod*=nums[r]
           
            while l < r and prod >=k:
                prod/=nums[l]
                l+=1
            if prod < k:
                c+=(r-l+1)
            
        return c
        


        